storm-core/src/jvm/org/apache/storm/trident/operation/OperationAwareFlatMapFunction.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.trident.operation;
+
+/**
+ * A one to many transformation function which is aware of Operation (lifecycle of the Trident component)
+ */
+public interface OperationAwareFlatMapFunction extends FlatMapFunction, Operation {
```

Comment:
Cant we directly modify the FlatMapFunction to implement Operation ? You could add default methods for prepare and cleanup and keep the interface Functional.